### ISSUE TITLE
http server fails to reply back on malformed requests

### ISSUE THREAD

**Issue by maumai** [Microsoft/cpprestsdk#31]
This error is related to bug https://casablanca.codeplex.com/workitem/267

A easy way to reproduce this is to run the BlackJack _Server and send "http://localhost:34568/blackjack/dealer?^"  to the server. Where the "?^" is the malformed part. The server should reply with status 400 "provided uri is invalid: /blackjack/dealer?^"

The reply message is not sent from some reason, this might have implications on server stability over time.

**Comment by kavyako** [comment]
Strange, the codeplex workitem mentions that this has been fixed in release 2.6.0. 
Let me look into this and see what happened to the fix.

**Comment by maumai** [comment]
I think it was fixed back in 2.6.0 but there is another issue that causes failing reply. Probably the same issue as described here: https://github.com/Microsoft/cpprestsdk/issues/13 
A correction for this was merged into development 7th Dec -15. And testing this on the development branch, now return correct response. 

Is it possible to get a corrected 2.7.1 release?.

**Comment by kavyako** [comment]
Thanks for confirming that the issue is fixed with the latest sources.

Unless there is a critical issue, we do not plan to make point releases. This will be included in our next release 2.8 (currently under planning). You can build the library from the latest sources on development branch to unblock your scenarios.

Thanks
Kavya

**Comment by maumai** [comment]
Ok.

It might we worth to have another look at the correction for https://github.com/Microsoft/cpprestsdk/issues/13. It still seems to fail to return a response message sometimes, for example if the error_code is set below:

void windows_request_context::read_headers_io_completion(DWORD error_code, DWORD)
{
    if(error_code != NO_ERROR)
    {
        m_msg.reply(status_codes::InternalError);
        init_response_callbacks(ShouldWaitForBody::DontWait); // <- does not cause a reply to client

**Comment by ras0219-msft** [comment]
Closing this for now; if there's a concrete repro of a current bug, please open a new issue.
